src/hotspot/share/jfr/instrumentation/jfrEventClassTransformer.cpp

```diff
@@ -909,9 +913,9 @@ static void adjust_stack_map(JfrBigEndianWriter& writer,
     writer.write<u1>(stream.get_u1(THREAD));
   }
 
-  u4 stack_map_attrib_len = writer.current_offset() - stack_map_attrib_len_offset;
+  u4 stack_map_attrib_len = static_cast<u4>(writer.current_offset() - stack_map_attrib_len_offset);
   // the stack_map_table_attributes_length value is exclusive
-  stack_map_attrib_len -= sizeof(u4);
```

Comment:
I'm thinking we should have globalDefinitions.hpp:

```
  const int u4_size = (int)sizeof(u4);
  const int u2_size = (int)sizeof(u2);
  const int u1_size = (int)sizeof(u1);
```

I don't think you have to static_cast<u4>(4) here.

Reply:
Since 4 is an int by default, the cast is needed.

Reply:
A constant value is fine.